src/client.js

```diff
@@ -129,97 +125,69 @@ Client.prototype.connect = function(args) {
 	var config = Helper.getConfig();
 	var client = this;
 
+	var nick = args.nick || "lounge-user";
+
+	var network = new Network({
+		name: args.name || "",
+		host: args.host || "",
+		port: parseInt(args.port, 10) || (args.tls ? 6697 : 6667),
+		tls: !!args.tls,
+		password: args.password,
+		username: args.username || nick.replace(/[^a-zA-Z0-9]/g, ""),
+		realname: args.realname || "The Lounge User",
+		commands: args.commands
+	});
+
+	client.networks.push(network);
+	client.emit("network", {
+		network: network
+	});
+
 	if (config.lockNetwork) {
 		// This check is needed to prevent invalid user configurations
 		if (args.host && args.host.length > 0 && args.host !== config.defaults.host) {
-			var invalidHostnameMsg = new Msg({
-				type: Msg.Type.ERROR,
-				text: "Hostname you specified is not allowed."
-			});
 			client.emit("msg", {
-				msg: invalidHostnameMsg
+				chan: network.channels[0].id,
+				msg: new Msg({
+					type: Msg.Type.ERROR,
+					text: "Hostname you specified is not allowed."
+				})
 			});
 			return;
 		}
 
-		args.host = config.defaults.host;
-		args.port = config.defaults.port;
-		args.tls = config.defaults.tls;
+		network.host = config.defaults.host;
+		network.port = config.defaults.port;
+		network.tls = !!config.defaults.tls;
```

Comment:
I don't believe it was there before since the affected line was `args.tls = config.defaults.tls`. Ideally, `config` should already have correct types though (so this sort of coercion, if necessary, should be done in the config helper, or be responsible for stopping execution when misconfigured).
We don't do checks for `config.defaults`, let's keep it that way for this PR.